zhaquirks/tuya/ts0601_motion.py

```diff
@@ -115,99 +219,141 @@ class NeoMotionManufCluster(TuyaNewManufCluster):
-    # presence_state: presence
-    # target distance: 1.61m
-    # illuminance: 250lux
-    # sensitivity: 9
-    # minimum_detection_distance: 0.00m
-    # maximum_detection_distance: 4.05m
-    # dp_detection_delay: 0.1
-    # dp_fading_time: 5.0
-    # ¿illuminance?: 255lux
-    # presence_brightness: no control
-    # no_one_brightness: no control
-    # current_brightness: off
+class TuyaMmwRadarClusterBase(NoManufacturerCluster, TuyaMCUCluster):
+    """Mmw radar cluster, base class."""
 
     attributes = TuyaMCUCluster.attributes.copy()
     attributes.update(
         {
             # ramdom attribute IDs
-            0xEF02: ("dp_2", t.uint32_t, True),
-            0xEF03: ("dp_3", t.uint32_t, True),
-            0xEF04: ("dp_4", t.uint32_t, True),
-            0xEF06: ("dp_6", t.enum8, True),
-            0xEF65: ("dp_101", t.uint32_t, True),
-            0xEF66: ("dp_102", t.uint32_t, True),
-            0xEF67: ("dp_103", t.CharacterString, True),
-            0xEF69: ("dp_105", t.enum8, True),
-            0xEF6A: ("dp_106", t.enum8, True),
-            0xEF6B: ("dp_107", t.enum8, True),
-            0xEF6C: ("dp_108", t.uint32_t, True),
+            0xEF01: ("occupancy", t.uint32_t, True),
+            0xEF02: ("sensitivity", t.uint32_t, True),
+            0xEF03: ("min_range", t.uint32_t, True),
+            0xEF04: ("max_range", t.uint32_t, True),
+            0xEF06: ("self_test", TuyaMmwRadarSelfTest, True),
+            0xEF09: ("target_distance", t.uint32_t, True),
+            0xEF65: ("detection_delay", t.uint32_t, True),
+            0xEF66: ("fading_time", t.uint32_t, True),
+            0xEF67: ("cli", t.CharacterString, True),
+            0xEF68: ("illuminance", t.uint32_t, True),
         }
     )
 
+
+class TuyaMmwRadarClusterV1(TuyaMmwRadarClusterBase):
+    """Mmw radar cluster, variant 1."""
+
     dp_to_attribute: Dict[int, DPToAttributeMapping] = {
         1: DPToAttributeMapping(
             TuyaOccupancySensing.ep_attribute,
             "occupancy",
         ),
         2: DPToAttributeMapping(
-            TuyaMCUCluster.ep_attribute,
-            "dp_2",
+            TuyaMmwRadarSensitivity.ep_attribute,
+            "present_value",
         ),
         3: DPToAttributeMapping(
-            TuyaMCUCluster.ep_attribute,
-            "dp_3",
+            TuyaMmwRadarMinRange.ep_attribute,
+            "present_value",
+            endpoint_id=2,
         ),
         4: DPToAttributeMapping(
-            TuyaMCUCluster.ep_attribute,
-            "dp_4",
+            TuyaMmwRadarMaxRange.ep_attribute,
+            "present_value",
+            endpoint_id=3,
         ),
         6: DPToAttributeMapping(
             TuyaMCUCluster.ep_attribute,
-            "dp_6",
+            "self_test",
         ),
         9: DPToAttributeMapping(
-            TuyaAnalogInput.ep_attribute,
+            TuyaMmwRadarTargetDistance.ep_attribute,
             "present_value",
             lambda x: x / 100,
         ),
```

Comment:
(continuing from previous/above comment:)
Here, the `present_value` on the "fake" `AnalogInput` cluster is updated with the target distance?

Is there a difference between this and the `target_distance` attribute in `TuyaMmwRadarClusterBase`?

Reply:
See my [other comment](https://github.com/zigpy/zha-device-handlers/pull/2525/files#r1368052764), but the `AnalogInput` sensor only gets created in ZHA for one manufacturer from what I can see: [sensor.py#L188-L198](https://github.com/home-assistant/core/blob/4ee9a6f1301d34c2f287038ebf0c3acdc5f93517/homeassistant/components/zha/sensor.py#L188-L198)

The correct way to create a sensor for this would be by having the Tuya datapoint map to a (fake) ZCL attribute and then exposing that through ZHA with something like this: [sensor.py#L1017-L1027](https://github.com/home-assistant/core/blob/4ee9a6f1301d34c2f287038ebf0c3acdc5f93517/homeassistant/components/zha/sensor.py#L1017-L1027)